### DISCUSSION TITLE
What is needed to re-install the toolkit suite on a new/replacement cluster without using `tk bootstrap`?

### DISCUSSION THREAD

**Discussion by shanesveller** [fluxcd/flux2#175] [Q&A] [answered]
For purposes of disaster recovery, migrating contents between clusters, onboarding new clusters in a more controlled way, etc. I would like to understand what are the minimum viable commands to stand up a toolkit from an existing "fleet" repo against a blank cluster. I'd like an option to do so without using `tk bootstrap` but instead assembling the workflow from deliberately controlled parts. My loose guess is that this is probably possible today a combination of `tk install` and manually creating the first secret for Git auth, similar to any other authenticated/private GitRepository? Plus making sure that SSH key is able to clone the repo in question, of course, and maybe manually defining the first GitRepository and a Kustomization CRD to pull it in.

This is motivated in part by the situation I faced on https://github.com/fluxcd/toolkit/issues/169#issuecomment-679324287.

**Comment by stefanprodan** [comment, marked as answer]
The manual bootstrap procedure is documented here https://toolkit.fluxcd.io/guides/installation/#generic-git-server

**Reply by shanesveller** [reply]
That looks very satisfactory, thank you!
